src/execution_plan/execution_plan.c

```diff
@@ -337,25 +346,25 @@ static void _ExecutionPlanSegment_ProcessQueryGraph(ExecutionPlanSegment *segmen
 
 		if(connectedComponentsCount > 1) {
 			// Connect traversal operations.
-			OpBase *childOp;
-			OpBase *parentOp;
-			Vector_Pop(traversals, &parentOp);
+			OpBase *parentOp = array_pop(traversals);
 			// Connect cartesian product to the root of traversal.
 			ExecutionPlan_AddOp(cartesianProduct, parentOp);
```

Comment:
This call is a bit confusing considering `ExecutionPlan_AddOp` signature
`ExecutionPlan_AddOp(OpBase *parent, OpBase *newOp)`